pkg/buses/monitor.go

```diff
@@ -805,13 +806,14 @@ func (m *Monitor) createOrUpdateBus(bus *channelsv1alpha1.Bus) error {
 }
 
 func (m *Monitor) createOrUpdateClusterBus(clusterBus *channelsv1alpha1.ClusterBus) error {
-	if clusterBus.Name != m.bus.GetObjectMeta().GetName() {
+	if m.bus.GetObjectKind().GroupVersionKind().Kind != clusterBus.Kind ||
```

Comment:
Does this mean we were sometimes passing a `ClusterBus` to `createOrUpdateBus` and vice-versa?

Reply:
`createOrUpdateClusterBus` will only receive a ClusterBus and `createOrUpdateBus` will only receive a Bus. The issue is that they both assign to `m.bus` which is a GenericBus interface that can be either.

The monitor is listening for Bus/ClusterBus so that it can receive updated parameters for channels and subscriptions. Realistically, it should only ever need to listen for either Buses or ClusterBuses, never both.

Reply:
Oh, this is a method on the monitor; I keep forgetting that these controllers aren't stateless (and each `Monitor` will see all the `Bus` & `ClusterBus` updates and choose which ones apply to its stored bus).

/lgtm
/approve